packages/python-packages/apiview-gpt/src/_gpt_reviewer.py

```diff
@@ -29,50 +31,70 @@ class GptReviewer:
-            input_variables=["apiview", "guidelines", "language"],
+            input_variables=["apiview", "guidelines", "language", "extra_comments", "class_list"],
             partial_variables={"format_instructions": self.output_parser.get_format_instructions()},
             template="""
-                Given the following {language} Azure SDK Guidelines:
-                  {guidelines}
-                Verify whether the following code satisfies the guidelines:
+                You are trying to analyze an API for {language} to determine whether it meets the SDK guidelines. We only provide one class at a time right now, but if you need it, here's a list of all the classes in this API:
+                {class_list}
+                
+                Here is the code for a single class:
                 ```
-                  {apiview}
+                {apiview}
                 ```
+
+                Identify any violations of the following guidelines:
+                {guidelines}
                 
+                Consider the following comments as well:
+                {extra_comments}
+
+                Format the output according to the following:
                 {format_instructions}
             """
         )
         self.chain = LLMChain(llm=self.llm, prompt=self.prompt_template)
 
     def get_response(self, apiview, language):
         apiview = self.unescape(apiview)
-        all_guidelines = self.retrieve_guidelines(language)
+        class_list = self.get_class_list(apiview)
 
-        # TODO: Make this not hard-coded!
-        guidelines = self.select_guidelines(all_guidelines, [
-            "python_design.html#python-client-naming",
-            "python_design.html#python-client-options-naming",
-            "python_design.html#python-models-async",
-            "python_design.html#python-models-dict-result",
-            "python_design.html#python-models-enum-string",
-            "python_design.html#python-models-enum-name-uppercase",
-            "python_design.html#python-client-sync-async",
-            "python_design.html#python-client-async-keywords",
-            "python_design.html#python-client-separate-sync-async",
-            "python_design.html#python-client-same-name-sync-async",
-            "python_design.html#python-client-namespace-sync",
-        ])
-
-        for i, g in enumerate(guidelines):
-            g["number"] = i
+        all_guidelines = self.retrieve_guidelines(language)
 
         chunked_apiview = SectionedDocument(apiview.splitlines(), chunk=True)
 
         final_results = GuidelinesResult(status="Success", violations=[])
         for chunk in chunked_apiview.sections:
             if self.should_evaluate(chunk):
-                results = self.chain.run(apiview=str(chunk), guidelines=guidelines, language=language)
+                # retrieve the most similar comments to identify guidelines to check
+                semantic_matches = VectorDB().search_documents(language, chunk)
```

Comment:
for now, are we just taking all the matches? edit: nvm I see we're using the defaults currently